seam/paginator.py

```diff
@@ -0,0 +1,102 @@
+from typing import Callable, Dict, Any, Tuple, Generator, List
+
+
+class Pagination:
+    def __init__(
+        self,
+        has_next_page: bool,
+        next_page_cursor: str | None,
+        next_page_url: str | None,
+    ):
+        self.has_next_page = has_next_page
+        self.next_page_cursor = next_page_cursor
+        self.next_page_url = next_page_url
+
+
+class Paginator:
+    """
+    Handles pagination for API list endpoints.
+
+    Iterates through pages of results returned by a callable function.
+    """
+
+    _FIRST_PAGE = "FIRST_PAGE"
+
+    def __init__(self, callable_func: Callable, params: Dict[str, Any] = None):
```

Comment:
I see we did this for `seam.createPaginator` in this and the PHP SDK. Can we do it for `Paginator` here as well and in the PHP SDK?
```suggestion
    def __init__(self, request: Callable, params: Dict[str, Any] = None):
```

Reply:
d844cf613323f09bfaa4a160b5a809ded26735a2